tests/test_detectAndMeasure.py

```diff
@@ -217,6 +265,261 @@ def _check_diaSource(self, refSources, diaSource, refIds=None,
+        config = detectAndMeasure.DetectAndMeasureScoreTask.ConfigClass()
+        config.doApCorr = False
+        config.doMerge = False
+        config.doSkySources = False
+        config.doForcedMeasurement = True
+        task = detectAndMeasure.DetectAndMeasureScoreTask(config=config)
+        output = task.run(science, matchedTemplate, difference, score)
+
+        def _check_values(values, minValue=None, maxValue=None):
+            self.assertTrue(np.all(np.isfinite(values)))
+            if minValue is not None:
+                self.assertTrue(np.all(values > minValue))
+            if maxValue is not None:
+                self.assertTrue(np.all(values < maxValue))
+        for column in columnNames:
+            _check_values(output.diaSources[column])
+        _check_values(output.diaSources.getX(), minValue=0, maxValue=xSize)
+        _check_values(output.diaSources.getY(), minValue=0, maxValue=ySize)
+        _check_values(output.diaSources.getPsfInstFlux())
+
+    def test_detect_transients(self):
+        """Run detection on a difference image containing transients.
+        """
+        noiseLevel = 1.
+        staticSeed = 1
+        transientSeed = 6
+        fluxLevel = 500
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+        config = detectAndMeasure.DetectAndMeasureScoreTask.ConfigClass()
+        config.doApCorr = False
+        config.doMerge = False
+        config.doSkySources = False
+        config.doForcedMeasurement = False
+        detectionTask = detectAndMeasure.DetectAndMeasureScoreTask(config=config)
+        subtractTask = subtractImages.AlardLuptonPreconvolveSubtractTask()
+        scienceKernel = science.psf.getKernel()
+
+        def _detection_wrapper(polarity=1):
+            transients, transientSources = makeTestImage(seed=transientSeed, psfSize=2.4,
+                                                         nSrc=10, fluxLevel=1000.,
+                                                         noiseLevel=noiseLevel, noiseSeed=8)
+            difference = science.clone()
+            difference.maskedImage -= matchedTemplate.maskedImage
+            if polarity > 0:
+                difference.maskedImage += transients.maskedImage
+            else:
+                difference.maskedImage -= transients.maskedImage
+            score = subtractTask._convolveExposure(difference, scienceKernel, subtractTask.convolutionControl)
+            output = detectionTask.run(science, matchedTemplate, difference, score)
+            refIds = []
+            for diaSource in output.diaSources:
+                self._check_diaSource(transientSources, diaSource, refIds=refIds, scale=polarity)
+        _detection_wrapper(polarity=1)
+        _detection_wrapper(polarity=-1)
```

Comment:
Polarity is really a bool, no? `False`=>negative, `True`->positive? Or just make it `positive=True/False` instead?